common/src/main/java/org/apache/comet/vector/CometVector.java

```diff
@@ -88,7 +90,11 @@ public Decimal getDecimal(int i, int precision, int scale) {
     if (!useDecimal128 && precision <= Decimal.MAX_INT_DIGITS() && type instanceof IntegerType) {
       return createDecimal(getInt(i), precision, scale);
     } else if (precision <= Decimal.MAX_LONG_DIGITS()) {
-      return createDecimal(useDecimal128 ? getLongDecimal(i) : getLong(i), precision, scale);
+      if (useDecimal128) {
+        return createDecimal(getLongFromDecimalBytes(getBinaryDecimal(i)), precision, scale);
```

Comment:
Hmm not sure how this is working with dictionary...

Reply:
`CometDictionaryVector` overrides `getBinaryDecimal` so we get the correct value in the buffer.

Reply:
I think `getBinaryDecimal ` calls `decodeToBinary` that copies buffer... I am curious whether it may slows down things for dictionaries...

Reply:
This PR leaves `CometDictionary.decodeToBinary` untouched (mainly because dictionary for decimals is a very rare case). However, `CometDictionary.decodeToBinary` does the same copy that `CometVector` was doing by calling `CometVector.copyBinaryValues` which this PR improves, so one would expect better performance? 
I haven't validated performance with dictionary values, though I have checked that the unit tests  pass (`ParquetReadSuite` covers dictionary encoded decimals).

Reply:
Thanks, I will look into this more to get better understanding.
The current `getLongDecimal` does not call `decodeToBinary` so I feel this change has more copies...

Reply:
Oh, right. I did not look into your `getLongDecimal` change. That might be better. Let me also verify.